### ISSUE TITLE
`HighlightStyle.Hover` etc... needs a different implementation 

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#3753]
Renamed this issue from "`HighlightOnPress` and/or `WantContinousButtonPressed` has issues with nested subviews." because there are deeper issues. 

The current impl changes the View's Scheme. Instead Highlight should be a `VisualRole`. 

I am NOT fixing this in #4062 , but instead disabling Hover all together until it can be done right in a PR for this Issue.

## Orignial Post

In #3751 I refactored and fixed a bunch of issues around mouse and highlight.

I'm not going to fix this there, so I'm creating an issue for it:

In the new `Mouse` scenario below each view in the `Enter/Leave Demo` is the same class... nested subviews:

- The root
  - One in the root's padding: inPadding
  - Subview: Sub1 
  - Subview: Sub2

Right now `WantContinousButtonPressed` is true for ALL 4. But `Highlight on Press` is only enabled for the root.

Note how sub1 and sub2 are showing highlight when I press in the Viewport of the root? that should not be happening.

![PQPZSes 1](https://github.com/user-attachments/assets/53c67745-0a48-43f7-abcb-489326cf4597)

**Comment by BDisp** [comment]
Yes it's true, having many highlight effects at once, hurting the eyes.

**Comment by tig** [comment]
> Yes it's true, having many highlight effects at once, hurting the eyes.

This is a demo intended to show all the highlights.

That's not what this issue is for.
